pootle/apps/pootle_translationproject/models.py

```diff
@@ -258,8 +271,24 @@ def save(self, *args, **kwargs):
         super(TranslationProject, self).save(*args, **kwargs)
 
         if created:
+            if initialize_from_templates:
+                # We are adding a new TP and there are no files to import from
+                # disk, so initialize the TP files using the templates TP ones.
+                from pootle_app.project_tree import init_store_from_template
+
+                for template_store in template_tp.stores.iterator():
+                    init_store_from_template(self, template_store)
+
             self.scan_files()
 
+            if initialize_from_templates:
+                # Trigger stats refresh for TP added from UI.
+                for store in self.stores.live().iterator():
+                    try:
+                        store.update()
+                    except Exception:
```

Comment:
> What kind of exception can you get here?

No idea, but whatever comes from https://github.com/translate/pootle/blob/878a1418/pootle/apps/pootle_store/models.py#L1547 for example.